src/main/java/main/TaskList.java

```diff
@@ -0,0 +1,179 @@
+        this.taskList = tasks;
+        this.numOfTasks = numOfTasks;
+    }
+
+    /**
+     * Return the number of tasks in the list.
+     *
+     * @return A number representing the number of all current tasks.
+     */
+    public int size() {
+        return numOfTasks;
+    }
+
+    /**
+     * Return the task at a specific position in the list.
+     *
+     * @param index The position of the queried task.
+     * @return Return the queried task.
+     */
+    public Task get(int index) {
+        return this.taskList.get(index);
+    }
+
+    /**
+     * Return a TaskList of tasks that happen on a specific date.
+     *
+     * @param date Date on which the task happens.
+     * @return The list of tasks filtered.
+     */
+    public TaskList filterByDate(Date date) {
+        List<Task> tasks = new ArrayList<Task>();
+        for (Task task : taskList) {
+            if (task.isOn(date)) {
+                tasks.add(task);
+            }
+        }
+        return new TaskList(tasks, tasks.size());
+    }
+
+    /**
+     * Return a new TaskList of tasks filtered by the given keyword.
+     *
+     * @param keyword The keyword to filter all match task description.
+     * @return A TaskList instance containing all tasks containing the keyword.
+     */
+    public TaskList filterByKeyword(String keyword) {
+        List<Task> tasks = new ArrayList<Task>();
+        for (Task task : taskList) {
+            if (task.isMatch(keyword)) {
+                tasks.add(task);
+            }
+        }
+        return new TaskList(tasks, tasks.size());
+    }
+
+    /**
+     * Add a task into the list of tasks.
+     *
+     * @param tNew A task to be added into the TaskList.
+     */
+    public void addTask(Task tNew) {
+        this.numOfTasks += 1;
+        this.taskList.add(tNew);
+    }
+
+    /**
+     * Return the task that has been deleted.
+     *
+     * @param index The index of the task to be deleted.
+     * @return The removed task to be printed out by Ui.
+     */
+    public Task deleteTask(int index) {
+        this.numOfTasks -= 1;
+        return this.taskList.remove(index - 1);
+    }
+
+    /**
+     * Return the task that has been marked as done.
+     *
+     * @param index The index of the task to be marked.
+     * @return The marked task to be printed out by Ui.
+     */
+    public Task markAsDone(int index) {
+        Task done = this.taskList.get(index - 1);
+        done.markAsDone();
+        return done;
+    }
+
+    /**
+     * Return the task that has been marked as undone.
+     *
+     * @param index The index of the task to be marked.
+     * @return The marked task to be printed out by Ui.
+     */
+    public Task markAsUndone(int index) {
+        Task undone = this.taskList.get(index - 1);
+        undone.markAsUndone();
+        return undone;
+    }
+
+    /**
+     * Return a string representation of the list of all tasks to be printed out.
+     * @return A string representation of the TaskList.
```

Comment:
Maybe you can put @\return on a newline instead? You can get Intellij to do [auto-formatting](https://stackoverflow.com/questions/17879475/how-enable-auto-format-code-for-intellij-idea) for you.
<img width="377" alt="Screenshot 2022-02-03 at 8 14 02 PM" src="https://user-images.githubusercontent.com/61443625/152340762-fbaa532b-b84e-4e2f-88be-73711512e1ec.png">

Reply:
Thanks for your useful trick!